### PR TITLE
Milestone two

### PR DESCRIPTION
**Pull request by Gracetemitope**
In this PR, I implemented add and remove functionality

### CODE REVIEW

File: src/components/AddBook.js [hunk 1]
```diff
@@ -1,23 +1,61 @@
...
+    e.preventDefault();
+    const newBook = {
+      item_id: uuid(), 
+      ...state,
+    }
+    dispatch(addBookRequest(newBook));
+    setState({ title: '', category: '' });
+    categoryRef.current.value = '';
+    titleRef.current.value = '';
+  }
```
Comment (dasileker) [inline src/components/AddBook.js:28]: - [x] The dispatch func is not working, Please fix it

File: src/redux/books/books.js [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+// Redux Saga Generators
+function* addBook({ payload }) {
+  try {
+    yield call(sendData, `${BASE_URL}/apps/${appID}/books`, payload);
+    yield put(addBookSuccess({ ...payload }));
+  } catch (err) {
+    console.log(err);
+  }
+}
```
Comment (dasileker) [inline src/redux/books/books.js:50]: - [ ] Also this function is not a async and await to fetch from the API. please fix

File: src/redux/books/books.js [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+function* getBooksWatcher() {
+  yield takeLatest(GET_BOOKS_REQUEST, getBooks);
+}
+
+export function* bookSagas() {
+  yield spawn(addBookWatcher)
+  yield spawn(removeBookWatcher)
+  yield spawn(getBooksWatcher)
+}
```
Comment (dasileker) [inline src/redux/books/books.js:87]: - [x] Also I can't remove anything the function is not working.

File: src/components/AddBook.js (hunk 1 above)
Comment (dasileker) [inline src/components/AddBook.js:28]: - [ ] You should have in the List of your book the remove btn & update.

**Review by dasileker — CHANGES_REQUESTED** [review]
### **Hi @Gracetemitope,** 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/books.js (hunk 2 above)
Comment (Gracetemitope) [reply src/redux/books/books.js:50]: This is not even a requirement for this milestone. I had finished before I realized.

**Review by AshakaE — APPROVED** [review]
### Hi @Gracetemitope  🙋‍♀️, 

# STATUS APPROVED ✔
Your project is complete!  There is nothing else to say other than... it's time to merge it 🔗 🚢 
Congratulations! 🎉 

Cheers 🥂  and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**